verification/verifier_helpers.go

```diff
@@ -176,3 +229,59 @@ func verifyX509TrustedIdentities(certs []*x509.Certificate, trustPolicy *TrustPo
 
 	return fmt.Errorf("signing certificate from the digital signature does not match the X.509 trusted identities %q defined in the trust policy %q", trustedX509Identities, trustPolicy.Name)
 }
+
+func (v *Verifier) executePlugin(ctx context.Context, trustPolicy *TrustPolicy, capabilitiesToVerify []plugin.VerificationCapability, signerInfo *nsigner.SignerInfo) (*plugin.VerifySignatureResponse, error) {
+	verificationPluginName := signerInfo.SignedAttributes.VerificationPlugin
+	var attributesToProcess []string
+	extendedAttributes := make(map[string]interface{})
+
+	for _, attr := range signerInfo.SignedAttributes.ExtendedAttributes {
+		extendedAttributes[attr.Key] = attr.Value
+		if attr.Critical {
+			attributesToProcess = append(attributesToProcess, attr.Key)
+		}
+	}
+
+	var certChain [][]byte
+	for _, cert := range signerInfo.CertificateChain {
+		certChain = append(certChain, cert.Raw)
+	}
+
+	signature := plugin.Signature{
+		CriticalAttributes: plugin.CriticalAttributes{
+			ContentType:        string(signerInfo.PayloadContentType),
+			SigningScheme:      string(signerInfo.SigningScheme),
+			Expiry:             &signerInfo.SignedAttributes.Expiry,
+			ExtendedAttributes: extendedAttributes,
+		},
+		UnprocessedAttributes: attributesToProcess,
+		CertificateChain:      certChain,
+	}
+
+	policy := plugin.TrustPolicy{
+		TrustedIdentities:     trustPolicy.TrustedIdentities,
+		SignatureVerification: capabilitiesToVerify,
+	}
+
+	request := &plugin.VerifySignatureRequest{
+		ContractVersion: plugin.ContractVersion,
+		Signature:       signature,
+		TrustPolicy:     policy,
+		PluginConfig:    ctx.Value("plugin-config").(map[string]string),
```

Comment:
The `Verifier.Verify()` method documentation says `plugin-context`, not `plugin-config`, one of them is wrong :P

On the other hand, context value keys should not use a default type (like string). This is to avoid having multiple packages, from different developers, overwrites each other keys.

I would do something like this:

```go
type pluginConfigCtxKey struct{}

// Used by callers to set the plugin config in the context.
func WithPluginConfig(ctx context.Context, config map[string]string) context.Context {
  return context.WithValue(ctx, pluginConfigCtxKey{}, config)
}

// getPluginConfig used by us to retrieve the config from the context.
func getPluginConfig(ctx context.Context, config map[string]string) map[string]string {
  // !! notice that I check that the conversion succeeded by using to arguments.
  // else it will panic if the context does not contain a config.
  config, ok := ctx.Value(pluginConfigCtxKey{}).(map[string]string)
  if !ok {
    return nil
  }
  return config
}
 
```